pynxtools/dataconverter/helpers.py

```diff
@@ -164,9 +164,9 @@ def generate_template_from_nxdl(
         ):
             template[optionality][f"{path}/@units"] = None
 
-        parent_path = convert_data_converter_dict_to_nxdl_path(path.rsplit("/", 1)[0])
+        nxdl_path = convert_data_converter_dict_to_nxdl_path(path)
         list_of_children_to_add = get_all_defined_required_children(
-            parent_path, nxdl_name
+            nxdl_path, nxdl_name
```

Comment:
I'm a bit lost. Can you provide me a way to quickly test this or describe what the prob was. 

The changes above make sense. But then I would've kept the parent_path here and passed that to add_inherited_children because we want to add all other inherited children of the parent of the current field/attribute being processed. Not of the field/attribute itself, no?

Reply:
The problem here is that this was passing the field `/ENTRY[entry]/definition` as path, which resulted in a `parent_path = /ENTRY` and then I got all of the required children of the entry but I just want to have all required children of `definition`.

That's also how I would expect it, no? I want to have the required children of the actual field not of its parent (this is also how it is done for the group below). Or do I misunderstand how this part of the code is working?

Reply:
I also sent you some data on how to test this on slack